src/main/java/gov/nih/nci/evs/api/controller/ConceptController.java

```diff
@@ -2140,6 +2142,90 @@ public class ConceptController extends BaseController {
     }
   }
 
+  /**
+   * Returns the codes for a given terminology.
+   *
+   * @param terminology the terminology
+   * @return the terminology codes
+   * @throws Exception the exception
+   */
+  @Operation(summary = "Get codes for the specified terminology")
+  @ApiResponses({
+    @ApiResponse(
+        responseCode = "200",
+        description = "Successfully retrieved the requested information",
+        content = @Content(schema = @Schema(implementation = ArrayList.class))),
+    @ApiResponse(
+        responseCode = "404",
+        description = "Resource not found",
+        content =
+            @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = RestException.class))),
+    @ApiResponse(
+        responseCode = "417",
+        description = "Expectation failed",
+        content =
+            @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = RestException.class)))
+  })
+  @Parameters({
+    @Parameter(
+        name = "terminology",
+        description = "Terminology, e.g. 'ncit'",
+        required = true,
+        schema = @Schema(implementation = String.class),
+        example = "ncit"),
+    @Parameter(
+        name = "X-EVSRESTAPI-License-Key",
+        description =
+            "Required license information for restricted terminologies. <a"
+                + " href='https://github.com/NCIEVS/evsrestapi-client-SDK/blob/master/doc/LICENSE.md'"
+                + " target='_blank'>See here for detailed information</a>.",
+        required = false,
+        schema = @Schema(implementation = String.class))
+  })
+  @RecordMetric
+  @GetMapping(value = "/concept/{terminology}/codes", produces = "application/json")
+  public @ResponseBody ArrayList<String> getAllCodesForTerminology(
```

Comment:
For this you should return List<String> - there's no reason to return the implementation type, the interface is better option generally